### ISSUE TITLE
Idea of faster translation: STT->TTS

### ISSUE THREAD

**Issue by matyaskopp** [ufal/charles-translator-android#21]
This is just an idea for improvement. 
It would be nice to have one button for both actions: 
- text to speech button
- play translated text

Like a small simulation of a human translator, who listens to a sentence, then translates it in mind and then says it.

**Comment by martinpopel** [comment]
Yes, this is the Dialog feature. I think, we should have both in the future, similarly to e.g. Google Translate app:

- single-mic icon "Hlas": just ASR, so the users can decide whether to tap the "play" icon in the translation or not
- double-mic icon "Konverzace": the Dialog feature, each translation is automatically played. Furthermore, there is an option "Automatický hlasový vstup", so that it tries to detect both languages and decide automatically the translation direction. I can imagine that making this feature reliable is a lot of work (uk-cs Google Translate automatic dialog is very unreliable, almost unusable in all my attempts so far).

> text to speech button

I think you meant "speech to text", i.e. ASR, i.e. "mic" button

**Comment by matyaskopp** [comment]
> I think you meant "speech to text", i.e. ASR, i.e. "mic" button

yes, the "mic" button

**Comment by krabatom** [comment]
Google has api for it https://developer.android.com/reference/android/speech/SpeechRecognizer and there is example of use https://medium.com/voice-tech-podcast/android-speech-to-text-tutorial-8f6fa71606ac But it is more comlicated to implement it than dialog. Maybe in future.

**Comment by stranak** [comment]
I want to leave the discussion of various aspects of "conversation mode" to #47, but a question specifically for this "auto-play" function in my mind is: Is this limited to the conversation mode, or does it make sense in the current, one-off translation mode too? 

I translate something, the phone reads the translation. I think some users might appreciate it. Could it be made a on/off switch somewhere in the UI (settings), something like "auto-play translations"? I think it could apply to both modes, the current one and the future "conversation mode" (#47).

**Comment by matyaskopp** [comment]
> something like "auto-play translations"

yes, this is what I exactly wanted, but I was not able to find the right words for "auto-play translation".

I can imagine that "conversation mode" can be difficult to implement.

So I am suggesting implementing **walkie-talkie mode**:
 - use two icons for the auto-play microphone (one for Czech and one for Ukraine) 
 - when you press the Czech-flagged auto-play symbol:
   - record Czech speech
   - translate it into Ukraine
   - play translation
 - when you press the Ukraine-flagged auto-play symbol:
   - record Ukraine speech
   - ...
